src/ActionList/List.tsx

```diff
@@ -162,15 +164,14 @@ export const List = React.forwardRef<HTMLDivElement, ListProps>((props, forwarde
   const renderItem = (itemProps: ItemInput, item: ItemInput, itemIndex: number) => {
     // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
     const ItemComponent = ('renderItem' in itemProps && itemProps.renderItem) || props.renderItem || Item
-    const key = itemProps.key ?? itemProps.id?.toString() ?? itemIndex.toString()
+    const key = itemProps.id?.toString() ?? itemIndex.toString()
     return (
       <ItemComponent
         showDivider={props.showItemDividers}
         selectionVariant={props.selectionVariant}
         {...itemProps}
         key={key}
         sx={{...itemStyle, ...itemProps.sx}}
-        item={item}
```

Comment:
@dgreif after my changes, `tsc` was complaining about this line, and I honestly couldn't figure out what the intent was here. Any ideas? Is it safe to remove?

Reply:
It's used by the callback handlers for onAction if I remember correctly.  This comes back to the object equivalence issue for selection.  We couldn't just pass an item "like" the one that was selected, we need the actual item that caused the `ItemComponent` to get rendered in the first place.  That whole system has made a lot of things more complicated then I would like them to be 😅

Reply:
Ohh that makes sense. I was wondering about that callback too. Any idea why the item in the callback was necessary?

Reply:
(that is, I can't picture the usage where this is helping simplify things...why wouldn't the user just keep their item in the closure of the callback?)

Reply:
I think the item in the callback was for the same reason.  Higher components listening to the callback need the exact item to store which thing got selected